L1Trigger/Phase2L1GMT/interface/Constants.h

```diff
@@ -0,0 +1,1293 @@
+#ifndef PHASE2GMT_CONSTANTS
+#define PHASE2GMT_CONSTANTS
+
+#include "ap_int.h"
+
+namespace Phase2L1GMT {
+
+  const int BITSCURV = 15;
+  const int BITSPT = 13;
+  const int BITSPHI = 13;
+  const int BITSETA = 13;
+  const int BITSZ0 = 10;
+  const int BITSD0 = 12;
+  const int BITSQUALITY = 4;
+
+  const int BITSSTUBCOORD = 8;
+  const int BITSSTUBETA = 8;
+  const int BITSSTUBID = 9;
+  const int BITSSTUBPHIQUALITY = 4;
+  const int BITSSTUBETAQUALITY = 4;
+  const int BITSSTUBTIME = 8;
+
+  const int BITSMATCHQUALITY = 9;
+  const int BITSSIGMACOORD = 4;
+  const int BITSSIGMAETA = 4;
+
+  // Bitwidth for standalone muons to CL1
+  const int BITSBX = 3;
+  const int BITSSAZ0 = 5;
+  const int BITSSAD0 = 7;
+  const int BITSSABETA = 4;
+
+  const int BITSTTCURV = 15;
+  const int BITSTTCURV2 = 27;
+  const int BITSTTPHI = 12;
+  const int BITSTTTANL = 16;
+  const int BITSTTZ0 = 12;
+  const int BITSTTD0 = 13;
+  const int BITSTTCHI2 = 4;
+  const int BITSTTBENDCHI2 = 3;
+  const int BITSTTMASK = 7;
+  const int BITSTTTRACKMVA = 3;
+  const int BITSTTOTHERMVA = 6;
+
+  const int BITSSTAMUONQUALITY = 6;
+  const int BITSMUONQUALITY = 4;
+  const int BITSMUONBX = 4;
+  const int BITSMUONBETA = 4;
+
+  const int BITSPROPCOORD = 9;
+  const int BITSPROPSIGMACOORD_A = 5;
+  const int BITSPROPSIGMACOORD_B = 5;
+  const int BITSPROPSIGMAETA_A = 5;
+  const int BITSPROPSIGMAETA_B = 5;
+
+  const float maxCurv_ = 0.00855;  // 2 GeV pT Rinv is in cm
+  const float maxPhi_ = 1.026;     // relative to the center of the sector
+  const float maxTanl_ = 8.0;
+  const float maxZ0_ = 30.;
+  const float maxD0_ = 15.4;
+  const int barrelLimit0_ = 181 / 4;
+  const int barrelLimit1_ = 160 / 4;
+  const int barrelLimit2_ = 140 / 4;
+  const int barrelLimit3_ = 110 / 4;
+  const int barrelLimit4_ = 0;
+
+  inline uint64_t twos_complement(long long int v, uint bits) {
+    uint64_t mask = (1 << bits) - 1;
+    if (v >= 0)
+      return v & mask;
+    else
+      return (~(-v) + 1) & mask;
+  }
+
+  template <class T>
+  inline int wordconcat(T word, int bstart, long int input, int bitsize) {
```

Comment:
I suppose `T word` was supposed to be `T & word` , otherwise I can't see how it would work.